python/lsst/ts/MTAlignment/alignmentModel.py

```diff
@@ -0,0 +1,507 @@
+        self.com_lock = asyncio.Lock()
+
+    async def connect(self):
+        """
+        Connect to the T2SA host
+        """
+        self.reader, self.writer = await asyncio.open_connection(self.host, self.port)
+        self.connected = True
+        asyncio.create_task(self.status_loop())
+
+    async def status_loop(self):
+        """
+        pings T2SA for status every second, and updates internal state.
+        """
+        while self.connected:
+            stat = await self.status()
+            if stat == "READY":
+                self.trackerstatus = TrackerStatus.READY
+            elif stat == "2FACE":
+                self.trackerstatus = TrackerStatus.TWOFACE
+            elif stat == "ADM":
+                self.trackerstatus = TrackerStatus.ADM
+            elif stat == "DRIFT":
+                self.trackerstatus = TrackerStatus.DRIFT
+            elif stat == "EMP":
+                self.trackerstatus = TrackerStatus.EMP
+            elif stat == "ERR":
+                self.trackerstatus = TrackerStatus.ERR
+            await asyncio.sleep(1)
+
+    async def wait_for_ready(self):
+        """
+        Checks if the tracker is doing something, if so, sleep until it's done.
+        """
+        with self.com_lock:
+            wait_states = ("EMP\r\n")
+            msg = bytes("?STAT\r\n", 'ascii')
+            self.writer.write(msg)
+            await self. writer.drain()
+            data = await  self.reader.read(64)
+            stat = data.decode()
+            if stat == "INIT" or stat == "INIT\r\n":
+                self.log.debug("waiting for init")
+                await asyncio.sleep(5)
+            while stat in wait_states:
+                self.log.debug("...")
+                await asyncio.sleep(0.3)
+                self.writer.write(msg)
+                await self. writer.drain()
+                data = await  self.reader.read(64)
+                stat = data.decode()
+            await asyncio.sleep(0.5)
+
+    async def send_msg(self, msg):
+        """
+        Formats and sends a message to T2SA.
+        """
+        msg = msg + "\r\n"
+        if type(msg) == str:  # this may move
+            msg = bytes(msg, 'ascii')
+        self.log.debug(f"sending {msg}")
+        with self.com_lock:
+            self.writer.write(msg)
+            await self. writer.drain()
+            data = await self.reader.readuntil(separator=bytes("\n", 'ascii'))
+            self.log.debug(f'Received: {data.decode()!r}')
+            return data.decode()
+
+    async def status(self):
+        """
+        query T2SA for status
+        """
+        response = await self.send_msg("?STAT")
+        return response
+
+    async def laser_status(self):
+        """
+        Query tracker laser status, and update model state accordingly
+        """
+
+        msg = self.msgformat("?LSTA")
+        data = await self.send_msg(msg)
+        if data == "LNC":
+            self.laserstatus = LaserStatus.LASERNOTCONNECTED
```

Comment:
Instead of storing the result in the class, consider returning the value. I think it is much more reliable to keep the model stateless....